test/gen-template/matter/endpoint-config.zapt

```diff
@@ -37,6 +37,13 @@
 
 #define ZAP_GENERATED_COMMANDS_INDEX(index) ((chip::CommandId *) (&generatedCommands[index]))
 
+// clang-format off
+#define GENERATED_EVENT_COUNT {{ chip_endpoint_generated_event_count }}
+#define GENERATED_EVENTS {{ chip_endpoint_generated_event_list }}
+// clang-format on
+
+#define ZAP_GENERATED_EVENTS_INDEX(index) ((chip::EventId *) (&generatedEvents[index]))
```

Comment:
Why do we need a cast here?  I don't think we should have a cast here....  We should just have our types be actually right.  Why don't we?

Reply:
I know we have it for commands, but we should not have it there either, either here or on the Matter side.

Reply:
Yes, it should work without casting. Just copied if from the line above for `generatedCommands`